openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,551 @@
+
+        # Text input to change the distance of reach when searching for nearest data points
+        self.scatter_distance = TextInput(value="0.1", title="Scatter Distance")
+        self.scatter_distance.on_change('value', self.scatter_input)
+        self.dist_range = float(self.scatter_distance.value)
+        
+        # Grouping all of the sliders and dropdowns into one column
+        sliders = [i for i in self.slider_dict.values()]
+        self.sliders_and_selects = row(
+            column(*sliders, self.x_input,
+            self.y_input, self.output_select, self.scatter_distance)
+        )
+
+        # Layout creation
+        self.layout = row(self.contour_data(), self.right_plot(), self.sliders_and_selects)
+        self.layout2 = row(self.bot_plot())
+        curdoc().add_root(self.layout)
+        curdoc().add_root(self.layout2)
+        curdoc().title = 'MultiView'
+
+    def make_predictions(self, data):
+
+        """
+        Runs the data parameter through the surrogate model which is given in prob.
+
+        Parameters
+        ----------
+        data : dict
+            Dictionary containing Ordered Dict of training points.
+
+        Returns
+        -------
+        array
+            np.stack of predicted points.
+        """
+        
+        outputs = {i : [] for i in self.output_list}
+        print("Making Predictions")
+
+        # Parse dict into shape [n**2, number of inputs] list
+        inputs = np.empty([self.resolution**2, self.nx])
+        for idx, values in enumerate(data.values()):
+            inputs[:, idx] = values.flatten()
+
+        # Pair data points with their respective prob name. Loop to make predictions
+        for idx, tup in enumerate(inputs):
+            for name, val in zip(data.keys(), tup):
+                self.prob[self.surrogate_comp.name + '.' + name] = val
+            self.prob.run_model()
+            for i in self.output_list:
+                outputs[i].append(float(self.prob[self.surrogate_comp.name + '.' + i]))
+
+        return self.stack_outputs(outputs)
+
+    def contour_data(self):
+
+        """
+        Creates a contour plot.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        Bokeh Image Plot
+        """
+
+        n = self.resolution
+        xe = np.zeros((n, n, self.nx))
+        ye = np.zeros((n, n, self.ny))
+
+        # Query the slider dictionary, append the name and current value to the ordered dictionary
+        self.slider_value_and_name = OrderedDict()
+        for title, slider_params in self.slider_dict.items():
+            self.slider_value_and_name[title] = slider_params.value
+        
+        # Cast the current values of the slider_value_and_name dictionary values to a list
+        self.input_point_list = list(self.slider_value_and_name.values())
+        for ix in range(self.nx):
```

Comment:
this looks like it could probably be done more quickly with a full broadcast, rather than a for loop